exercises/diffie-hellman/src/test/java/DiffieHellmanTest.java

```diff
@@ -0,0 +1,95 @@
+import org.junit.Before;
+import org.junit.Ignore;
+import org.junit.Test;
+
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.assertEquals;
+
+import java.math.BigInteger;
+import java.util.*;
+
+public class DiffieHellmanTest {
+
+    private DiffieHellman diffieHellman;
+
+    @Before
+    public void setUp() {
+        diffieHellman = new DiffieHellman();
+    }
+
+    @Test
+    public void testPrivateKeyInRange() {
+        BigInteger primeP = BigInteger.valueOf(23);
+        List<BigInteger> privateKeys = new ArrayList<>();
+        for (int i = 0; i < 10; i++) privateKeys.add(diffieHellman.privateKey(primeP));
+
+        for (BigInteger privateKey : privateKeys) {
+            assertTrue((privateKey.compareTo(BigInteger.ONE) >= 0) && (privateKey.compareTo(primeP) < 0));
```

Comment:
I would split this into two assertions. That makes it more readable and it makes it easy for the user to see exactly what's failing if the test fails 🙂

Reply:
I agree, just splitted into two assertions 🙂